bolt/lib/Core/BinaryContext.cpp

```diff
@@ -2523,6 +2523,16 @@ BinaryFunction *BinaryContext::getBinaryFunctionAtAddress(uint64_t Address) {
   return nullptr;
 }
 
+/// Deregister JumpTable registered at a given \p Address and delete it.
+void BinaryContext::deleteJumpTable(uint64_t Address) {
+  JumpTable *JT = getJumpTableContainingAddress(Address);
+  assert(JT && "Must have a jump table at address");
```

Comment:
Maybe assert that the jump table you're trying to delete is also _exactly_ at Address? If I understand correctly the logic here, you don't expect the jump table to start at any other address other than Address. If it does, we are accidentally deleting something that we shouldn't be, no?

Or do you want to cover cases where the reference is not exactly to Address as well?